tf-modules/archive/main.tf

```diff
@@ -12,4 +12,11 @@ locals {
     aws_security_group.no_ingress_all_egress[0].id,
     var.elasticsearch_security_group_id
   ])
+  all_non_internal_buckets = [for k, v in var.buckets : v.name if v.type != "internal"]
```

Comment:
Is this too permissive? Allowing the Lambda to access all buckets except "internal" buckets?

Reply:
It looks like for the `lambda_processing` role used by our Step function tasks, we just allow access to every bucket in `buckets`:

https://github.com/nasa/cumulus/blob/master/tf-modules/cumulus/iam.tf#L83

So now I almost wonder if we should do the same here? Especially since `system_bucket` often **is an internal bucket**

Reply:
In theory this increases the permissions footprint.  I'm fine either way so long as the CHANGELOG notes the change, but if I had to choose, I'd opt to YAGNI that concern, given there's no good reason for the API to need access to a secondary `internal` bucket that isn't the system bucket.